packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -743,6 +743,33 @@ abstract class DiagnosticsNode {
 
   String get _separator => showSeparator ? ':' : '';
 
+  /// Serialize the node excluding its descendents to a JSON map.
+  ///
+  /// Subclasses should override if they have additional properties that are
+  /// useful for the GUI tools that consume this JSON.
+  ///
+  /// See also:
+  ///
+  ///  * [WidgetInspectorService], which forms the bridge between JSON returned
+  ///    by this method and interactive tree views in the Flutter IntelliJ
+  ///    plugin.
+  @mustCallSuper
+  Map<String, Object> toJson() {
```

Comment:
this should be toMap or something. In doesn't return JSON. (the data type for JSON data is String.)

Reply:
Are you ok with toJsonMap?
There are many examples of other Dart APIs containing methods called toJson where the return type is of type Map. For example: https://github.com/google/streamy-dart/blob/master/lib/runtime/request.dart#L129
As the return of the method is typed there is no ambiguity whether the return is a String with encoded json or a Map with decoded JSON.

I would be concerned the name toMap does not convey the concept that the Map better be valid JSON so should stick to the subset of the language that can be encoded as json.